hibernate-core/src/main/java/org/hibernate/proxy/map/MapLazyInitializer.java

```diff
@@ -53,4 +53,9 @@ protected boolean isAllowLoadOutsideTransaction() {
 	protected String getSessionFactoryUuid() {
 		return super.getSessionFactoryUuid();
 	}
+
+	@Override
+	protected String getSessionFactoryName() {
```

Comment:
I might be wrong, but this method seems unnecessary as it's overriding the super method with the same code?
If I misunderstood, please add a comment.

Reply:
Actually I thought the same initially ;-) but without it, it gives:

```
> Task :hibernate-core:compileJava
hibernate-orm/hibernate-core/src/main/java/org/hibernate/proxy/map/MapProxy.java:123: error: getSessionFactoryName() has protected access in AbstractLazyInitializer
  li.getSessionFactoryName(),    
```

Note that `MapLazyInitializer` also has similar:

```
	@Override
	protected String getSessionFactoryUuid() {
		return super.getSessionFactoryUuid();
	}
```

It needs to be there as `MapProxy` uses them (same package as `MapLazyInitializer`!) but is not extending any of the classed mentioned above but only using `MapLazyInitializer` internally, and hence the latter one needs to expose it.